pkg/plugin/api/api.go

```diff
@@ -55,8 +55,13 @@ func (a *grpcAPI) Start(ctx context.Context) error {
 		service: a.Service,
 	}
 
+	size := viper.GetInt("client-max-recv-msg-size")
+	if size == 0 {
```

Comment:
If we want to prevent zero as a size, then we should do it in the flag setup. I believe we can set min max thresholds

Reply:
If `api.Start` is called from a test environment where flags are not setup, setting a default earlier would not be read unless the size is passed down rather than fetched through `GetInt`. I ended up setting a viper default in the test although we will need to remember to pass the size explicitly instead if there is ever a use case for creating `NewDashboardClient` externally.